src/test/java/seedu/address/commons/util/CsvUtilTest.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.commons.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.PostalCode;
+import seedu.address.model.tag.Tag;
+
+
+/**
+ * Unit tests for CsvUtil.
+ */
+public class CsvUtilTest {
+
+    private ObservableList<Person> personList;
+
+    @BeforeEach
+    public void setUp() {
+        // Create actual Person objects for the test
+        Set<Tag> tags1 = new HashSet<>();
+        tags1.add(new Tag("Friend"));
+        tags1.add(new Tag("Colleague"));
+
+        Set<Tag> tags2 = new HashSet<>();
+        tags2.add(new Tag("Family"));
+
+        // Update addresses to ensure door numbers have #
+        Person person1 = new Person(
+                new Name("Alice"),
+                new Phone("82345678"),
+                new Email("alice@example.com"),
+                new Address("#12-345, 123 Wonderland"),
+                new PostalCode("118424"),
+                tags1
+        );
+
+        Person person2 = new Person(
+                new Name("Bob"),
+                new Phone("97654321"),
+                new Email("bob@example.com"),
+                new Address("#23-456, 456 Nowhere"),
+                new PostalCode("118424"),
+                tags2
+        );
+
+        personList = FXCollections.observableArrayList(person1, person2);
+    }
+
+    @Test
+    public void testConvertObservableListToCsv() {
```

Comment:
It might be better to follow the naming standard of the test cases across the rest of the codebase to ensure consistency